implementations/rust/ockam/ockam_command/src/space/create.rs

```diff
@@ -35,61 +36,55 @@ pub struct CreateCommand {
     pub identity_opts: IdentityOpts,
 }
 
-impl CreateCommand {
-    pub fn run(self, opts: CommandGlobalOpts) -> miette::Result<()> {
-        async_cmd(&self.name(), opts.clone(), |ctx| async move {
-            self.async_run(&ctx, opts).await
-        })
-    }
-
-    pub fn name(&self) -> String {
-        "space crate".into()
-    }
+#[async_trait]
+impl Command for CreateCommand {
+    const NAME: &'static str = "space create";
 
-    async fn async_run(&self, ctx: &Context, opts: CommandGlobalOpts) -> miette::Result<()> {
+    async fn async_run(self, ctx: &Context, opts: CommandGlobalOpts) -> Result<()> {
         if !opts
             .state
             .is_identity_enrolled(&self.identity_opts.identity_name)
             .await?
         {
-            return Err(miette!(
-                "Please enroll using 'ockam enroll' before using this command"
-            ));
+            return Err(
+                miette!("Please enroll using 'ockam enroll' before using this command").into(),
+            );
         };
 
-        opts.terminal.write_line(format!(
-            "\n{}",
-            "Creating a trial space for you (everything in it will be deleted in 15 days) ..."
-                .light_magenta(),
-        ))?;
-        opts.terminal.write_line(format!(
-            "{}",
-            "To learn more about production ready spaces in Ockam Orchestrator, contact us at: hello@ockam.io".light_magenta()
-        ))?;
-
         let node = InMemoryNode::start(ctx, &opts.state).await?;
-        let space = node
-            .create_space(
+
+        let space = {
+            let pb = opts.terminal.progress_bar();
+            if let Some(pb) = pb.as_ref() {
+                pb.set_message("Creating a Space for you...");
+            }
+            node.create_space(
                 ctx,
                 &self.name,
                 self.admins.iter().map(|a| a.as_ref()).collect(),
             )
-            .await?;
-
+            .await?
+        };
+        if space.is_in_free_trial_subscription() {
+            opts.terminal.write_line(fmt_log!("This Space does not have a Subscription attached to it."))?
+                .write_line(fmt_log!("As a courtesy, we created a temporary Space for you, so you can continue to build.\n"))?
+                .write_line(fmt_log!("Please subscribe to an Ockam plan within two weeks {}", color_uri("https://www.ockam.io/pricing")))?
+                .write_line(fmt_log!("{}\n", color_warn("If you don't subscribe in that time, your Space and all Projects will be permanently deleted.")))?;
+        }
         opts.terminal
             .stdout()
             .plain(space.item()?)
-            .json(serde_json::json!(&space))
+            .json_obj(&space)?
```

Comment:
Nice shortcut.

Reply:
It's half-way there, but the idea is to remove the current `json` method and rename the `json_obj` to `json`. But to do that, we need first to have properly formatted outputs in every command.